spec/mailers/contacting_creator_mailer_spec.rb

```diff
@@ -1572,6 +1572,49 @@ def expect_push_alert(seller_id, text)
     end
   end
 
+  describe ".subscribers_data" do
+    let(:recipient) { create(:user) }
+    let(:filename) { "subscribers-export-#{SecureRandom.hex}.csv" }
+    let(:tempfile) { Tempfile.new }
+
+    before do
+      tempfile.puts "csv content"
+    end
+
+    it "contains the correct attachment and attributes" do
+      mail = ContactingCreatorMailer.subscribers_data(
+        recipient: recipient,
+        tempfile: tempfile,
+        filename: filename,
+      )
+
+      expect(mail.to).to eq([recipient.email])
+      expect(mail.subject).to include("Here is your subscribers data")
+      expect(mail.attachments.size).to eq(1)
+      expect(mail.attachments.first.body.raw_source).to eq("csv content\r\n")
+    end
+
+    context "when attachment size is above threshold" do
+      before do
+        stub_const("MailerAttachmentOrLinkService::MAX_FILE_SIZE", 1.byte)
+      end
+
+      it "contains a link instead of an attachment" do
+        mail = ContactingCreatorMailer.subscribers_data(
+          recipient: recipient,
+          tempfile: tempfile,
+          filename: filename,
+        )
+
+        expect(mail.to).to eq([recipient.email])
+        expect(mail.subject).to include("Here is your subscribers data")
+        expect(mail.attachments.size).to eq(0)
+        expect(mail.body).to include(filename)
+        expect(mail.body).to have_link("Please click this link")
```

Comment:
I think this is the only actual test failure - feel free to either:

1. Stub the download URL instead of `MailerAttachmentOrLinkService::MAX_FILE_SIZE`

https://github.com/antiwork/gumroad/blob/494b90486952e2fab975e41e8148cb28045ca21d/spec/mailers/contacting_creator_mailer_spec.rb#L1518-L1524

and then do something like
	
```ruby
expect(mail.body).to have_link("link", href: download_link)
```
	
2. Or, your original approach of Nokogiri or [`strip_tags`](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/SanitizeHelper.html#method-i-strip_tags).

Sorry I didn't realize that only the word "link" is in the `<a>`!